mllib/src/main/scala/org/apache/spark/mllib/clustering/KMeans.scala

```diff
@@ -521,6 +519,13 @@ object KMeans {
       v2: VectorWithNorm): Double = {
     MLUtils.fastSquaredDistance(v1.vector, v1.norm, v2.vector, v2.norm)
   }
+
+  private[spark] def validateInitializationMode(initializationMode: String): Boolean = {
```

Comment:
This should return true or false, not throw an exception.  The caller can throw an exception as needed.  (spark.ml Params handle throwing the exception if isValid fails.)